src/hotspot/os_cpu/bsd_aarch64/os_bsd_aarch64.cpp

```diff
@@ -372,10 +391,10 @@ JVM_handle_bsd_signal(int sig,
       Thread::WXWriteFromExecSetter wx_write;
 
       // Handle signal from NativeJump::patch_verified_entry().
-      if ((sig == SIGILL)
+      if ((sig == SIGILL || sig == SIGTRAP)
```

Comment:
sigtrap here is linux-only, we have removed it for purpose of it being wrong here